ui/js/services/jobfilters.js

```diff
@@ -407,15 +407,25 @@ treeherder.factory('thJobFilters', [
          */
         function getFieldFiltersArray() {
             const fieldFilters = [];
+            const clopt = thClassificationTypes.classificationOptions;
 
             _.each($location.search(), function (values, fieldName) {
                 if (_isFieldFilter(fieldName)) {
-                    const valArr = _toArray(values);
+                    var valArr = _toArray(values);
                     _.each(valArr, function (val) {
+                        var text = '';
+                        if (_withoutPrefix(fieldName) === 'failure_classification_id') {
+                            for (var i = 0; i < clopt.length; i++) {
+                                if (clopt[i]['id'].toString() === val) {
+                                    text = clopt[i]['name'];
+                                }
+                            }
+                        }
                         if (fieldName !== QS_SEARCH_STR) {
                             fieldFilters.push({
                                 field: _withoutPrefix(fieldName),
                                 value: val,
+                                text: text,
```

Comment:
So I opted to always add `text` to each element in `fieldFilters`, rather than only adding `text` when it's a failure classification filter. So for other types of field filters like job group, etc, etc, they'll simply have a `text = ''` property which seems harmless and not costly.

I did try the other approach of pushing `text` only to failure classification elements, and at least in the way I was attempting it, I was getting goofy results when applying multiple failure classification filters, or applying mixed filters. It seemed the above way was the simplest, and least brittle.

Reply:
And the new test L416-423 could also live inside `if (fieldName !== QS_SEARCH_STR) {` if you prefer. I assume that `if` is just some safety valve and returns true in most cases anyway. Maybe there's still some economy in moving it inside, not sure.

Reply:
This looks like a good approach to me.  :)